packages/devtools_app/lib/src/memory/flutter/memory_controller.dart

```diff
@@ -65,6 +67,55 @@ class MemoryController extends DisposableController
 
   String get memorySource => _memorySourceNotifier.value;
 
+  /// Starting chunk for slider based on pruneInterval.
+  double sliderValue = 1.0;
+
+  /// Number of interval chunks e.g., 5 minute interval has 3 chunks for 15 minutes of collected data
+  int numberOfChunks = 0;
+
+  /// Compute total timeline chunks used by Timeline slider.
+  int computeChunks() {
+    int chunks = 0;
+    if (memoryTimeline.data.isNotEmpty) {
+      final lastSampleTimestamp = memoryTimeline.data.last.timestamp.toDouble();
+      final firstSampleTimestamp =
+          memoryTimeline.data.first.timestamp.toDouble();
+      chunks = ((lastSampleTimestamp - firstSampleTimestamp) /
+              pruneIntervalDurationInMs)
+          .round();
+    }
+    return chunks == 0 ? 1 : chunks;
+  }
+
+  static const String displayOneMinute = '1';
+  static const String displayFiveMinutes = '5';
+  static const String displayTenMinutes = '10';
+  static const String displayAllMinutes = 'All';
+
+  /// Automatic pruning of memory statistics (plotted) full data is still retained.
+  /// Default is to display last minute of collected data in the chart.
+  final _pruneInterfaceNoifier = ValueNotifier<String>(displayOneMinute);
+
+  ValueListenable<String> get displayIntervalNotifier => _pruneInterfaceNoifier;
+
+  set pruneInterval(String interval) {
+    _pruneInterfaceNoifier.value = interval;
+  }
+
+  String get pruneInterval => _pruneInterfaceNoifier.value;
+
+  /// 1 minute in milliseconds.
+  static const int minuteInMs = 60 * 1000;
+
+  /// Largest int in ms for VM could be 9223372036854775807 (2^64)
+  /// however, use JS largest possible int  9007199254740991 (2^53).
+  static const int bigTimeInMs = 9007199254740991;
```

Comment:
see `maxJsInt` in utils. This is what we use elsewhere in devtools

Reply:
Done.